streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -151,6 +158,35 @@ else if (thisLoad > otherLoad)
             return capacity > other.capacity;
     }
 
+    boolean hasMoreAvailableCapacityThan(final ClientState other) {
+        checkCapacity(other);
+
+        final double otherLoad = (double) other.assignedTaskCount() / other.capacity();
+        final double thisLoad = (double) assignedTaskCount() / capacity();
+
+        return leastLoaded(thisLoad, otherLoad, other);
+    }
+
+    boolean hasMoreAvailableActiveTaskCapacityThan(final ClientState other) {
+        checkCapacity(other);
+
+        final double otherLoad = (double) ((other.numberOfActiveStateStores + 1) * other.numberOfActivePartitions
+                + other.activeTasks().size()) / other.capacity;
+        final double thisLoad = (double) ((numberOfActiveStateStores + 1) * numberOfActivePartitions
+                + activeTasks.size()) / capacity;
+
+        return leastLoaded(thisLoad, otherLoad, other);
+    }
+
+    boolean hasMoreAvailableStandbyTaskCapacityThan(ClientState other) {
```

Comment:
this name is ok as it is because standby tasks always have a state store associated with them.